app/components/top_controls.rb

```diff
@@ -84,6 +93,28 @@ def copy_text_button(bar)
     end
   end
 
+  def tashkeel_toggle_button(bar)
+    bar.tooltip(text: t(".hide_tashkeel_tooltip")) do
+      bar.button(
+        data: {
+          action: "click->top-controls#toggleTashkeel",
+          top_controls_target: "tashkeelToggleButton"
+        }
+      ) do
+        # Eye icon (default state - showing tashkeel)
+        Lucide::Eye(
+          class: "size-5 transition-all duration-200",
+          data: { top_controls_target: "tashkeelToggleIconEye" }
+        )
+        # EyeOff icon (hidden state - tashkeel hidden)
+        Lucide::EyeOff(
+          class: "size-5 transition-all duration-200 hidden",
+          data: { top_controls_target: "tashkeelToggleIconEyeOff" }
+        )
```

Comment:
Could you please follow the PhlexIcons gem documentation [here](https://github.com/AliOsm/phlex-icons?tab=readme-ov-file#add-custom-icons-to-your-rails-application) and add [these](https://gist.github.com/AliOsm/d41bb2e9379ed3af128b893d6ae2d1cd) 2 icons under the `Custom` module. You can convert HTML/SVG to Phlex components using https://phlexing.fun. I can make it, but I want you to enjoy the journey 😂